Sources/Kitura-CredentialsTwitter/CredentialsTwitter.swift

```diff
@@ -0,0 +1,358 @@
+    /// - inProgress: The closure to invoke to cause a redirect to the login page in the case of redirecting
+    ///       authentication.
+    func twitterRedirect(request: RouterRequest,
+                         response: RouterResponse,
+                         onFailure: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                         inProgress: @escaping () -> Void) {
+        do {
+            _ = try response.redirect("https://api.twitter.com/oauth/authenticate?oauth_token=\(oAuthToken)")
+            inProgress()
+        }
+        catch {
+            Log.error("Could not redirect to Twitter")
+            onFailure(nil, nil)
+        }
+    }
+
+    //MARK: Step 3: Converting the request token to an access token
+    
+    
+    /// The method gathers the necessary components to create a request to https://api.twitter.com/oauth/access_token.
+    /// At this point, a valid OAuth Token, OAuth Secret, and OAuth Verifier, gathered during Steps 1 and 2 should be 
+    /// valid. If the request is successful, it will move on to Step 2 which redirects the user. Failure or incorrect 
+    /// payloads call `onFailure` or `onPass`, respectively.
+    /// - Parameters:
+    ///   - request: The `RouterRequest` object used to get information about the request.
+    ///   - response: The `RouterResponse` object used to respond to the request.
+    ///   - onSuccess: The closure to invoke in the case of successful authentication.
+    ///   - onFailure: The closure to invoke in the case of an authentication failure.
+    ///   - onPass: The closure to invoke when the plugin doesn't recognize the authentication data (usually an
+    ///       authentication token) in the request.
+    func twitterAccessTokenRequest(request: RouterRequest,
+                                   response: RouterResponse,
+                                   onSuccess: @escaping (UserProfile) -> Void,
+                                   onFailure: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                                   onPass: @escaping (HTTPStatusCode?, [String:String]?) -> Void) {
+        var requestOptions: [ClientRequest.Options] = []
+        requestOptions.append(.schema("https://"))
+        requestOptions.append(.hostname("api.twitter.com"))
+        requestOptions.append(.method("POST"))
+        requestOptions.append(.path("oauth/access_token"))
+        var headers = [String : String]()
+        var parameters = [String:String]()
+
+        let nonce = String.nonce
+        let timeStamp = Date().timeIntervalSince1970.roundedString
+
+        parameters["oauth_consumer_key"] = consumerKey
+        parameters["oauth_nonce"] = nonce
+        parameters["oauth_signature_method"] = "HMAC-SHA1"
+        parameters["oauth_timestamp"] = timeStamp
+        parameters["oauth_token"] = oAuthToken
+        parameters["oauth_version"] = "1.0"
+        parameters["oauth_verifier"] = oAuthTokenVerifier
```

Comment:
I seriously considered it but I faulted on the side of having code that was more “followable” for someone coming in. I did think that having a baseline dictionary of parameters would suffice but wanted to keep it more approachable, if that makes sense.